### PR TITLE
Release/v1.0.4

### PR DESCRIPTION
**Pull request by huawenm**
*Issue #, if available:*
 * porting failed since no project references in the request
 * minimum  support version being changed.

*Description of changes:*
 * fixed project references in the porting
 * rollback SDK to support previous IDE versions 

*Testing done:*
add integration test coverage

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including READMEs and comments (where appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by flywanli — DISMISSED** [review]

**Review by marknfawaz — DISMISSED** [review]
Can we add "Microsoft.VisualStudio.SDK" to the list of ignored packages in dependabot? If already added, you can ignore.

File: tst/PortingAssistantExtensionIntegTests/PAIntegTestClient.cs [hunk 1]
```diff
@@ -152,7 +152,7 @@ public async Task<AnalysisTestResult> AssessSolutionAsync()
             return analysisResults;
         }
 
-        public async Task<AnalysisTestResult> PortSolutionAsync()
+        public async Task<AnalysisTestResult> PortSolutionAsync(bool includeCodeFix = false)
```
Comment (cslong) [inline tst/PortingAssistantExtensionIntegTests/PAIntegTestClient.cs:155]: Do you need update the request to use this new parameter?

File: src/PortingAssistantExtensionServer/Services/PortingService.cs [hunk 2]
```diff
@@ -93,6 +102,33 @@ public ProjectFilePortingResponse PortingProjects(ProjectFilePortingRequest requ
                 };
             }
         }
+
+        private List<RecommendedAction> GenerateRecommendedActions(ProjectFilePortingRequest request)
+        {
+            try
+            {
+                var upgradePacakgesResults = PackageToAnalysisResults
```
Comment (cslong) [inline src/PortingAssistantExtensionServer/Services/PortingService.cs:110]: can you fix the typo haha

Comment (huawenm) [reply src/PortingAssistantExtensionServer/Services/PortingService.cs:110]: thanks!

File: tst/PortingAssistantExtensionIntegTests/PAIntegTestClient.cs (hunk 1 above)
Comment (huawenm) [reply tst/PortingAssistantExtensionIntegTests/PAIntegTestClient.cs:155]: good question, there's a different between IDE and standalone tools, the standalone tools cache results from PA in the frontend but IDE extension cache the information in the language server. So the only thing we need from frontend are customer inputs (target framework and porting project path). the recommendation and project details are auto filled from cache if we have any.

**Review by cslong — APPROVED** [review]

**Review by kanikaul-amazon — APPROVED** [review]

**Pull request merged**